routes/orders.js

```diff
@@ -0,0 +1,71 @@
+const ordersRouter = require('express').Router()
+
+const {requireAdmin, requireUser} = require('./utils')
+
+const {
+    getAllOrders, 
+    getCartByUser, 
+    getOrderById, 
+    createOrder, 
+    getOrdersByUser
+} = require('../db')
+
+ordersRouter.get('/', requireAdmin ,async (req, res, next) => {
+    try {
+        const orders = await getAllOrders()
+        if(orders) {
+            res.send({orders})
+        }
+    } catch (error) {
+        next(error)
+    }
+})
+
+ordersRouter.get('/cart', requireUser, async (req, res, next) => {
+    try {
+        const order = await getCartByUser(req.user)
+        if (order){
+            res.send({order}) 
+        }
+    } catch (error) {
+        next(error)
+    }
+})
+
+ordersRouter.post('/', requireUser, async (req, res, next) => {
+    try {
+        const order = await createOrder(req.body)
+        if(order) {
+            res.send({order})
+        }
+    } catch (error) {
+        next(error)
+    }
+})
+
+ordersRouter.get('/:orderId', requireUser, async (req, res, next) => {
+    const {orderId} = req.params
+    try {
+        const order = await getOrderById(orderId)
+        if (order) {
+        res.send({order})
```

Comment:
REQUIRED - Only send back the order if it matches the user ID of the user who owns the order.  (matching the order.userId)

Reply:
Fixed

Reply:
For future reference, this'll have to be changed to allow if isAdmin is true. `if(order.userId === req.user.id || req.user.isAdmin)`